consensus/propagation/have_wants.go

```diff
@@ -343,6 +340,24 @@ func (blockProp *Reactor) handleRecoveryPart(peer p2p.ID, part *proptypes.Recove
 			return
 		}
 
+		// broadcast haves for all parts since we've decoded the entire block.
+		// rely on the broadcast method to ensure that parts are only sent once.
+		haves := &proptypes.HaveParts{
+			Height: part.Height,
+			Round:  part.Round,
+		}
+
+		for i := uint32(0); i < parts.Total(); i++ {
+			p, has := parts.GetPart(i)
+			if !has {
+				blockProp.Logger.Error("failed to get decoded part", "peer", peer, "height", part.Height, "round", part.Round, "part", i)
+				continue
+			}
+			haves.Parts = append(haves.Parts, proptypes.PartMetaData{Index: i, Proof: p.Proof, Hash: p.Proof.LeafHash})
+		}
+
+		blockProp.broadcastHaves(haves, peer, int(parts.Total()))
```

Comment:
addressed comment to broadcast all haves as we decode, therefore can ignore incoming haves if we have the full block. This will get hit if we have the full block. added a todo to stop re-sending haves and to test for it